cosmetics-web/app/controllers/my_account/email_controller.rb

```
@@ -16,8 +16,13 @@ def update
 
       @user.new_email_pending_confirmation!(dig_params(:new_email))
       render "users/check_your_email/show"
-    rescue StandardError
-      render :edit
+    rescue ActiveRecord::RecordInvalid => e
+      # We don't want to let the user know when validation failed due to new email being already registered
+      if e.message.include? "has already been taken"
```

Not very happy with this. 
We want to display an error on "edit" page when the user attempts to use a wrongly formatted email, while we want to fail silently and show the "check your email" page when the user attempts to use an email that already belongs to another user.
Didn't find a simpler/better way to separate both cases as the exception class is the same :man_shrugging:
Happy to get suggestions though.